pkg/reslife/container.go

```diff
@@ -0,0 +1,31 @@
+package reslife
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/justtrackio/gosoline/pkg/appctx"
+)
+
+type (
+	containerCtxKey struct{}
+)
+
+func AddLifeCycleer(ctx context.Context, factory LifeCycleerFactory) error {
+	var err error
+	var container *[]LifeCycleerFactory
+
+	if container, err = provideContainer(ctx); err != nil {
+		return fmt.Errorf("could not add life cycle purger: %w", err)
+	}
+
+	*container = append(*container, factory)
```

Comment:
Is it okay that this is not thread safe?

Reply:
locked